krun.py

```diff
@@ -172,6 +173,15 @@ def jobs_until_eta_known(self):
     def add_job(self, job):
         self.work_deque.append(job)
 
+    def remove_job_by_key(self, key):
+        for job in self.work_deque:
+            if job.key == key:
+                job_to_remove = job
+                break
+        else:
+            raise JobMissingFromConfig(key)
```

Comment:
Minor nit. `JobMissingFromConfig` is a misleading name, as jobs don't exist in the config atall? I would go for the more generic `JobMissingError` or seomthing?